### ISSUE TITLE
Plan for Action

### ISSUE THREAD

**Issue by Rusketh** [Rusketh/ExpAdv2#388]
Remove NextBot
Remove GUI Screens
Remove broken derma
Remove net messages
Remove url protection

Add new url protection
Add new net messages
Add npc entitys
Fix existing issues

Scrap everything and compete with star fall?

**Comment by szymski** [comment]
Compiler, Rusketh, compiler!!! You have to rewrite the compiler. Make it proper and readable. The current one doesn't even produce AST. This is wrong. No one except you can understand it. If you want EA2 to be fast, you have to split the compiler into multiple files/stages.
Look at this: https://github.com/szymski/QuantumChip/blob/master/docs/compiler_structure.md
Separate stages solve many problems, like including or debugging the compiler. I've never wanted to contribute to EA2's compiler, because its code has always scared me.
If the code looks like this https://github.com/szymski/QuantumChip/tree/master/luaa/quantumchip/compiler it's a pleasure to work on it. You should really consider remaking the compiler.
BTW, QuantumChip is dead for now. GMod has become boring for me and I couldn't find anyone to help me with QC. However, if you're willing to rewrite EA2's compiler, I'd be glad to help. I've always liked in-game scripting, but there isn't any good language out here. E2 is shit. EA2 is slow. Starfall uses Lua. Go ahead, read through my compiler and apply some techniques to new expadv2 compiler.
And unblock me on Steam, lmao.

**Comment by bigdogmat** [comment]
I have to agree with @szymski on this one, the compiler really needs to be rewritten. It's soo slow considering you're doing string concatenation for the entire thing.

**Comment by Rusketh** [comment]
If i rewrote the compiler i would go back to a vm stack like e2, compiling to native string is just...
Well it was a cool experiment, but it did not work.

**Comment by Rusketh** [comment]
The new compiler will have a tokenizer a parser and a memory validator but no actual compiler stage.  How? Because we know what we are doing this time round :D

**Comment by Xithyl515** [comment]
Yea I would love a new compiler too, I don't like waiting 60 seconds per compile or lagging the server to death.

**Comment by szymski** [comment]
Please let us know how the new compiler would work before you start and wait for our response. I don't want it to have any design mistakes. The more people know about it, the more problems will be found.

**Comment by Rusketh** [comment]
And why would I do that?
Once I have finished my current project, I will begin the new simpler compiler.

**Comment by Xithyl515** [comment]
No point in editing on here. Lol the email shows alllllllllll!!!! Anyways Rusketh If you need testers I got a code that stresses the compiler haha. Nah I'm sure you have stuff to test. Thank you though and I would love to have progress updates if you ever get around to it.
